src/main/java/com/cloudbees/plugins/deployer/engines/Engine.java

```diff
@@ -275,26 +272,16 @@ public FingerprintingWrapper(FilePath.FileCallable<DeployedApplicationLocation>
             this.listener = listener;
         }
 
-        public DeployedApplicationLocation invoke(File f, VirtualChannel channel)
+        public AbstractMap.SimpleEntry<String, DeployedApplicationLocation> invoke(File f, VirtualChannel channel)
                 throws IOException, InterruptedException {
             DeployedApplicationLocation location = delegate.invoke(f, channel);
             if (f.isFile() && location != null) {
                 FileInputStream fis = new FileInputStream(f);
                 try {
                     String md5sum = Util.getDigestOf(fis);
-                    FingerprintDecorator decorator = new FingerprintDecorator(listener, md5sum, location);
-                    Jenkins instance = Jenkins.getInstance();
-                    if (instance != null) {
-                        return decorator.call();
-                    } else {
-                        Channel current = Channel.current();
-                        assert current != null
-                                : "Can only have a null Jenkins instance if on the other end of a channel";
-                        return current.call(decorator);
-                    }
+                    return new AbstractMap.SimpleEntry<String, DeployedApplicationLocation>(md5sum, location);
                 } catch (Exception e) {
```

Comment:
The `try`-`catch` could probably be deleted and the caller made to handle any of the usual exceptions, as it probably already is. I suspect this was here in case the callback to master failed, which is now an obsolete condition.